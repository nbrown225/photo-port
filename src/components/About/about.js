import React from 'react';
import coverImage from '../../assets/cover/cover-image.jpg';

function About(){
  return(
    <section className='my-5'>
    <h1 id='about'>Who Am I?</h1>
    <img src={coverImage} className='my-2' style={{ width: '100%'}} alt='cover' />
    <div className='my-2'>
      <p>
        lorea adjgaen aevadnanaebaoed ao aaobaob aeoiga[oboervgaoaojaobab aodidonabankrnn oag aobnaooi oiOIF Ao VO IOgno [o ioiEOAOO n naonas oaoeijao oaeos aodn ooadnfoan oanv]]
      </p>
    </div>
  </section>
  )
}

export default About;