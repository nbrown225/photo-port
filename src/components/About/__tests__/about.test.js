import React from "react";
import { render, cleanup } from "@testing-library/react";

// import jest-dom extend-expect 
import '@testing-library/jest-dom/extend-expect';
// import component being tested 'About'
import About from "../about";

// call cleanup function / ensure no leftover memory / no false results
afterEach(cleanup);

// call describe function, to declare component being tested
describe('About component', () =>{
    //renders About test
    // first test / verifies if component is rendering
    it('renders', () =>{ //it can me used in place of test
        render(<About></About>)
    })

    // second test
    it('matches snapshot DOM node structure', () =>{
        // render about
        const { asFragment } = render(<About></About>)
        expect(asFragment()).toMatchSnapshot();
    })
})

