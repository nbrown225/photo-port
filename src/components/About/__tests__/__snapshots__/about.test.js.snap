// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="my-5"
  >
    <h1
      id="about"
    >
      Who Am I?
    </h1>
    <img
      alt="cover"
      class="my-2"
      src="cover-image.jpg"
      style="width: 100%;"
    />
    <div
      class="my-2"
    >
      <p>
        lorea adjgaen aevadnanaebaoed ao aaobaob aeoiga[oboervgaoaojaobab aodidonabankrnn oag aobnaooi oiOIF Ao VO IOgno [o ioiEOAOO n naonas oaoeijao oaeos aodn ooadnfoan oanv]]
      </p>
    </div>
  </section>
</DocumentFragment>
`;
